src/ServiceManagerBase.sol

```diff
@@ -87,25 +89,92 @@ abstract contract ServiceManagerBase is ServiceManagerBaseStorage {
     }
 
+    /**
+     * @notice Creates a new performance-based rewards submission, to be split amongst the operators and
+     * set of stakers delegated to operators who are registered to this `avs`.
+     * @param performanceRewardsSubmissions The performance rewards submissions being created.
+     * @dev Only callabe by the permissioned rewardsInitiator address
+     * @dev The duration of the `rewardsSubmission` cannot exceed `MAX_REWARDS_DURATION`
+     * @dev The tokens are sent to the `RewardsCoordinator` contract
+     * @dev This contract needs a token approval of sum of all `operatorRewards` in the `performanceRewardsSubmissions`, before calling this function.
+     * @dev Strategies must be in ascending order of addresses to check for duplicates
+     * @dev Operators must be in ascending order of addresses to check for duplicates.
+     * @dev This function will revert if the `performanceRewardsSubmissions` is malformed.
+     */
+    function createAVSPerformanceRewardsSubmission(
+        IRewardsCoordinator.PerformanceRewardsSubmission[]
+            calldata performanceRewardsSubmissions
+    ) public virtual onlyRewardsInitiator {
+        for (uint256 i = 0; i < performanceRewardsSubmissions.length; ++i) {
+            // Calculate total amount of token to transfer
+            uint256 totalAmount = 0;
+            for (
+                uint256 j = 0;
+                j < performanceRewardsSubmissions[i].operatorRewards.length;
+                ++j
+            ) {
+                totalAmount += performanceRewardsSubmissions[i]
+                    .operatorRewards[j]
+                    .amount;
+            }
+
+            // Transfer token to ServiceManager and approve RewardsCoordinator to transfer again
+            // in createAVSPerformanceRewardsSubmission() call
+            performanceRewardsSubmissions[i].token.transferFrom(
+                msg.sender,
+                address(this),
+                totalAmount
+            );
+            uint256 allowance = performanceRewardsSubmissions[i]
+                .token
+                .allowance(address(this), address(_rewardsCoordinator));
+            performanceRewardsSubmissions[i].token.approve(
+                address(_rewardsCoordinator),
+                totalAmount + allowance
+            );
+        }
+
+        _rewardsCoordinator.createAVSPerformanceRewardsSubmission(
+            address(this),
+            performanceRewardsSubmissions
+        );
+    }
+
+    /**
+     * @notice Forwards a call to Eigenlayer's RewardsCoordinator contract to set the address of the entity that can call `processClaim` on behalf of this contract.
+     * @param claimer The address of the entity that can call `processClaim` on behalf of the earner
+     * @dev Only callabe by the permissioned rewardsInitiator address
+     */
+    function setClaimerFor(
+        address claimer
+    ) public virtual onlyRewardsInitiator {
```

Comment:
`onlyOwner` instead

Reply:
Are we issuing rewards to the AVS in Rewards v2?

Reply:
There's an edge case in Performance-based rewards where if the AVS includes an `operator` in the submission but that operator wasn't registered to the AVS at any point during the specified `duration`. Then in that case the onchain transaction still goes through and the amount is deposited in the RewardsCoordinator. 

But during rewards calculation in the sidecar, it will have validation to check if that operator was registered during the duration (We can't do this check onchain since this isn't checkpointed and we can't look back historically currently - but we can offchain). If it wasn't registered, the entire rewards submission is sent back to the AVS as a payment leaf in the rewards merkle tree (which the AVS has to claim).

Reply:
Explained more in the `Security Considerations` section of the ELIP draft: https://docs.google.com/document/d/1nPl1TC0PD61a30MJVLjpF1kG_20pI2ki-AXWwqbScNk/edit?tab=t.0#heading=h.9jxq0ip86ron

Under `Preventing Rewards Distribution Tree bloat`

Reply:
Done in [3d99748](https://github.com/Layr-Labs/eigenlayer-middleware/pull/315/commits/3d9974848612772ac69b30b703648497e9c11bff)

Reply:
oh gotcha, I didn't realize that was happening via a leaf in the merkle tree 

sgtm